app/Http/Controllers/API/v1/CategoryController.php

```diff
@@ -43,16 +68,38 @@ class CategoryController extends ApiController
     )]
     public function index(Request $request): JsonResponse
     {
+        $last_synced = now();
         $user = $request->user();
         $categories = $user->categories();
+        $limit = 20;
+        if ($request->has('limit')) {
+            $limit = $request->limit;
+        }
         if ($request->has('type') && $request->type == 'income') {
             $categories = $categories->where('type', 'income');
         } elseif ($request->has('type') && $request->type == 'expense') {
             $categories = $categories->where('type', 'expense');
         }
-        $categories = $categories->paginate(20);
+        if ($request->has('sync_from')) {
```

Comment:
```suggestion
        if ($request->has('synced_from')) {
```


or something similar would be more intuitive, to show that we are talking about something that happened in the past.